core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -662,3 +672,54 @@ private Logger logger() {
         return DockerLoggerFactory.getLogger(COMPOSE_EXECUTABLE);
     }
 }
+
+@UtilityClass
+class DockerComposePrecondition {
+    private static final String CONTAINER_NAME_OPTION = "container_name";
+    private static final Yaml YAML = buildYamlParser();
+
+    static void checkNotContainUnsupportedComposeOptions(List<File> composeFiles) {
+        composeFiles.forEach(file -> checkArgument(!containUnsupportedComposeOptions(file),
+            String.format("Compose file %s contains '%s' option which is not supported by container.",
+                file,
+                CONTAINER_NAME_OPTION)));
+    }
+
+    private static boolean containUnsupportedComposeOptions(File file) {
+        return transformToYamlFile(file)
+            .map(DockerComposeYamlFile::getServices)
+            .filter(services -> !services.isEmpty())
+            .map(services -> services.values()
+                .stream()
+                .anyMatch(service -> service.containsKey(CONTAINER_NAME_OPTION)))
+            .orElse(false);
+    }
+
+    private static Optional<DockerComposeYamlFile> transformToYamlFile(File file) {
+        try {
+            return Optional.of(YAML.load(FileUtils.openInputStream(file)));
+        } catch (ConstructorException e) {
+            logger().warn(String.format("Unable to read yaml structure from compose file %s.", file.getName()), e);
+            return Optional.empty();
+        } catch (IOException e) {
+            throw new ContainerLaunchException(String.format("Unable to read compose file %s.", file.getName()), e);
+        }
+    }
+
+    private static Yaml buildYamlParser() {
+        Representer representer = new Representer();
+        representer.getPropertyUtils().setSkipMissingProperties(true);
+
+        return new Yaml(new Constructor(DockerComposeYamlFile.class), representer);
```

Comment:
`DockerComposeYamlFile` class is unnecessary, SnakeYAML can decode to `Map` directly (also helps to avoid the Representer